build/storage/tests/it/ptf_tests/system_tools/docker.py

```diff
@@ -0,0 +1,171 @@
+from system_tools.ssh_terminal import SSHTerminal
+
+
+class Docker:
+    def __init__(self, terminal):
+        self.terminal = terminal
+        self.containers = {}
+
+    def get_docker_containers_id_from_docker_image_name(self, docker_image_name):
+        out = self.terminal.execute(
+            f'sudo docker ps | grep "{docker_image_name}"'
+        ).splitlines()
+        return [line.split()[0] for line in out]
+
+    def _delete_all_containers(self):
+        """Delete all containers even currently running"""
+        out = self.terminal.execute("docker ps -aq")
+        if out:
+            self.terminal.execute("docker container rm -fv $(docker ps -aq)")
+
+    def _delete_running_containers(self):
+        """Delete running containers"""
+        for container in self.containers.values():
+            container.stop()
+
+    def delete_containers(self):
+        """Delete containers"""
+        return (
+            self._delete_all_containers()
+            if TestsConfig().debug.upper() == "FALSE"
+            else self._delete_running_containers()
+        )
+
+    def add_container(self, name, container_id):
+        self.containers[name] = container_id
+
+
+class DockerContainer:
+    def __init__(self, terminal, cmd_to_start, image_name):
+        self.tests_config = TestsConfig()
+        self.image_name = image_name
+        self._terminal = terminal
+        self._cmd_to_start = cmd_to_start
+        self.id = None
+        self.start()
+
+    def start(self):
+        out = self._terminal.execute(self._cmd_to_start)
+        self._wait_for_running(self.image_name)
+        self.id = out.split()[-1].strip("'")
+        return self.id
+
+    def stop(self):
+        self._terminal.execute(f"docker container rm -fv {self.id}")
+
+    @retry(stop=stop_after_delay(180), reraise=True)
+    def _wait_for_running(self, image_name):
+        out = self._terminal.execute("docker ps")
+        if image_name not in out:
+            raise ContainerNotRunningException
+
+
+class StorageTargetContainer(DockerContainer):
+    def __init__(self):
+        terminal = SSHTerminal(StorageTargetConfig())
+        cmd = (
+            f"cd {terminal.config.storage_dir} && "
+            f"AS_DAEMON=true scripts/run_storage_target_container.sh"
+        )
+        super().__init__(terminal, cmd, "storage-target")
+
+
+class IPUStorageContainer(DockerContainer):
+    def __init__(self):
+        shared_dir = HostTargetConfig().vm_share_dir_path
+        terminal = SSHTerminal(IPUStorageConfig())
+        cmd = (
+            f"cd {terminal.config.storage_dir} && "
+            f"AS_DAEMON=true SHARED_VOLUME={shared_dir} "
+            f"scripts/run_ipu_storage_container.sh"
+        )
+        super().__init__(terminal, cmd, "ipu-storage")
+
+
+class CMDSenderContainer(DockerContainer):
+    def __init__(self):
+        if not hasattr(self, "id"):
+            terminal = SSHTerminal(IPUStorageConfig())
```

Comment:
this container can be placed not only on ipu platform but on any others. Can we have this configurable by passing `BasePlatformConfig` into constructor?

Reply:
config as a parameter in init

Reply:
At this moment we have that config injected within this class
```
class KVMPlatformsFactory(PlatformsFactory):
    def __init__(self):
        self.cmd_sender = CMDSenderContainer(IPUStorageConfig())
```
But if we decide to run the tests on hw platform, we will have to create a dedicated class which will pass a required config.
I assume we should strive to make that step configurable.
What if we create a dedicated flag in our config, which will say where we need to run that cmd sender?
```
lass PlatformsFactory:
    def __init__(self):
        self.cmd_sender = None
        if config.cmd_sender_platform == "ipu":
            self.cmd_sender = CMDSenderContainer(IPUStorageConfig())
        elif config.cmd_sender_platform == "storage-target":
            self.cmd_sender = CMDSenderContainer(IPUStorageConfig())
        elif config.cmd_sender_platform == "host":
            self.cmd_sender = CMDSenderContainer(HostTargetConfig())
        else:
            # assert
```

Reply:
changed